File: Python/_496_Next_Greater_Element_I/solution.py
class Solution:
    def nextGreaterElement(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        mapping = {}
        for i in range(len(nums2)):
            for j in range(i, len(nums2)):
                if nums2[j] > nums2[i]:
                    mapping[nums2[i]] = nums2[j]
                    break
                if j == len(nums2) - 1:
                    mapping[nums2[i]] = -1
        res = []
        for i in range(len(nums1)):
            res.append(mapping[nums1[i]])
        return res

File: Java/_496_Next_Greater_Element_I/Solution.java
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
        Map<Integer, Integer> mapping = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums2.length; i++)
        {
            for (int j = i; j < nums2.length; j++)
            {
                if (nums2[j] > nums2[i])
                {
                    mapping.put(nums2[i], nums2[j]);
                    break;
                }
                if (j == nums2.length - 1)
                    mapping.put(nums2[i], -1);
            }
        }       
        int[] res = new int[nums1.length];
        for (int i = 0; i < nums1.length; i++)
            res[i] = mapping.get(nums1[i]);
        return res;
    }
}